record.go

```diff
@@ -0,0 +1,42 @@
+package kafka
+
+import (
+	"github.com/segmentio/kafka-go/protocol"
+)
+
+// Header is a key/value pair type representing headers set on records.
+type Header = protocol.Header
```

Comment:
> Type aliases are not meant for everyday use. They were introduced to support gradual code repair while moving a type between packages during large-scale refactoring.

This is the advice I always see about type aliases. That's now what we're doing here, so I think it's a code smell. Can you talk a bit about why type aliases are the right choice here?

Reply:
For example, RecordReader could alternately be defined like this:
```
type RecordReader interface {
    protocol.RecordReader
}
```
and so on.

Reply:
(or, presumably, the `protocol` types could be used directly everywhere they are needed)

Reply:
For the interfaces that's doable, but not for the concrete types, we'd have to redeclare them, which then means they're a different type and would not be compatible when assigning anymore (e.g. we can't assign a `kafka.Header` to `protocol.Header` without explicitly converting the values). I unified everything to use type aliases to avoid having two models.

We had `kafka.Header` before, with the same signature, which is why I used a type alias here. The main concern here is to keep the exported APIs as clean as possible, and avoid breaking compilation as much as possible. This code is very much a transition, we now have two implementations of the protocol in one repository, and more worked to do to simplify it. Type aliases were introduced to help with these large code refactors, so I think the use case is fine here, I expect that we'll evolve the APIs throughout the versions, and these type aliases will go away.

Regarding using the `protocol` package directly, I'd like to discourage programs from doing so, to give us the opportunity to make breaking changes to this package without breaking programs. Maybe we could make it `internal/protocol`, but some programs may be OK with breaking changes if it means they can use the lower level APIs, and the current design of `kafka.RoundTripper` kinds of require `protocol` to be exported.
I didn't want to bring all low-level protocol symbols in the top-level package either (which would have avoided the aliases), because it extends the exported API surface beyond my comfort zone.

If we were starting from scratch, we would probably have gotten a cleaner design, but retaining compatibility with existing code is important here.